bson/codec_options.py

```diff
@@ -201,9 +201,33 @@ def __eq__(self, other: Any) -> Any:
 
 class DatetimeConversionOpts(enum.IntEnum):
     DATETIME = 1
+    """Decode a BSON UTC datetime as a :class:`datetime.datetime`.
+
+    BSON UTC datetimes that cannot be represented as a
+    :class:`~datetime.datetime` will raise an :class:`~builtins.OverflowError`
```

Comment:
Is it always an OverflowError? I recall sometimes it's a ValueError.

Reply:
Yeah, I think it can looking at the source for `datetime.datetime`